arm-insights/2015-04-01/swagger/insightsManagementClient.json

```diff
@@ -0,0 +1,2119 @@
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets the tags of the rule."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AutoscaleSetting",
+          "description": "Gets or sets the setting to create or update."
+        }
+      },
+      "description": "Parameters supplied to the Create or Update autoscale setting operation."
+    },
+    "RetentionPolicy": {
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Gets or sets a value indicating whether the retention policy is enabled."
+        },
+        "days": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the number of days for the retention."
+        }
+      },
+      "required": [ "enabled", "days" ],
+      "description": "Part of MultiTenantDiagnosticSettings. Child of MetricSettings. Specifies the retention policy for the specific metric."
+    },
+    "LogProfile": {
+      "properties": {
+        "storageAccountId": {
+          "type": "string",
+          "description": "Gets or sets the resource id of the storage account."
+        },
+        "serviceBusRuleId": {
+          "type": "string",
+          "description": "Gets or sets the resource id of the service bus rule."
+        },
+        "locations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Gets or sets the locations."
+        },
+        "categories": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": " Gets or sets the categories."
+        },
+        "retentionPolicy": {
+          "$ref": "#/definitions/RetentionPolicy",
+          "description": "Gets or sets the retention policy for this log."
+        }
+      },
+      "description": "The log profile."
+    },
+    "LogProfileResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Gets or sets the resource identifier of the log profile."
+        },
+        "name": {
+          "type": "string",
+          "description": "Gets or sets the name of the log profile."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/LogProfile",
+          "description": "Gets or sets the properties of the log profile."
+        }
+      },
+      "description": "The log profile resource."
+    },
+    "LogProfileCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LogProfileResource"
+          },
+          "description": "Gets or sets the values of the log profiles."
+        }
+      },
+      "description": "Represents a collection of log profiles."
+    },
+    "TimeSpan": {
```

Comment:
This model definition should be deleted. Wherever "TimeSpan" is referred, please make the type as string and format as duration.
When I looked at the value of the timegrain property over here https://msdn.microsoft.com/en-us/library/azure/dn931939.aspx, it seems that format duration should be the correct thing to do.